lib/teleterm/gateway/kube.go

```diff
@@ -165,7 +174,10 @@ func (k *kube) makeKubeMiddleware() (alpnproxy.LocalProxyHTTPMiddleware, error)
 		// TODO(tross): update this when kube is converted to use slog.
 		Logger:       slog.Default(),
 		CloseContext: k.closeContext,
-	}), nil
+	})
+
+	k.clearCertsFn = middleware.ClearCerts
```

Comment:
I'm not 100% sure if this is going to work, but I think it might be better to create a `middleware` field on `kube` of interface `KubeMiddleware` that has just the `ClearCerts` method and then set that field in `makeALPNLocalProxyForKube` after creating the middleware.

I assume you couldn't use `alpnproxy.LocalProxy.SetCert` here because of how the kube proxy works?

Reply:
That's nicer, thanks.

> I assume you couldn't use alpnproxy.LocalProxy.SetCert here because of how the kube proxy works?

AFAIK kube proxies pass handling requests to the middleware:
https://github.com/gravitational/teleport/blob/41bae13c2f9eb07eebb5579eea4dbf142b335b76/lib/srv/alpnproxy/local_proxy.go#L406-L409

So I had to clear them there.